packages/fcl/src/exec/verify.js

```diff
@@ -66,25 +58,53 @@ const VERIFY_SIG_SCRIPT = `
   }
       
   pub fun main(
+    acctAddress: Address,
     message: String,
-    rawPublicKeys: [String],
-    weights: [UFix64],
-    signAlgos: [UInt],
-    hashAlgos: [UInt],
+    keyIds: [Int],
     signatures: [String],
   ): Bool {
 
     let keyList = Crypto.KeyList()
+    let rawPublicKeys: [String] = []
+    let weights: [UFix64] = []
+    let signAlgos: [UInt] = []
+    let hashAlgos: [UInt] = []
+    let uniqueKeys: {Int: Bool} = {}
```

Comment:
Could also define a `struct` in the script to hold all these values unique to each key, and have a single array of the struct instead of multiple arrays each with elements from each key. This approach is totally fine as well though

Reply:
Thanks, can refactor this on next update